shcomplete/repos.py

```diff
@@ -95,11 +94,14 @@ def fetch_repos(args):
     """
     Fetch the repositories and write the raw links into an output txt file
     """
-    g = GitHub(args.token, password=None, timeout=40, per_page=50)
-    repos = []
-    for cls in __shells__:
-        repos += cls.get_repos(g)
-    repos = to_raw_urls(repos)
+    g = GitHub(args.token, password=None, timeout=100, per_page=100)
+    all_repos = set()
+    for _ in range(5):
+        repos = set()
+        for cls in __shells__:
+            repos.update(cls.get_repos(g))
+        repos = to_raw_urls(repos)
+        all_repos.update(repos)
```

Comment:
Ok, I spent quite a lot of time to think how to do this properly.
The API works randomly.  Given the same parameters, it returned (after removing doubles) between 750 and 850 repos. These parameters appeared to give the best results on average.
Besides, doing a "cross query" like I do here (launching the same query 5 times) I obtained around 1250 repos which might be very close to the maximum.